src/textual/widgets/_digits.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+from typing import cast
+
+from rich.align import Align, AlignMethod
+from rich.console import RenderableType
+
+from ..geometry import Size
+from ..renderables.digits import Digits as DigitsRenderable
+from ..widget import Widget
+
+
+class Digits(Widget):
+    """A widget to display numerical values using a 3x3 grid of unicode characters.
+
+    Args:
+        value: Value to display in widget.
+        name: The name of the widget.
+        id: The ID of the widget in the DOM.
+        classes: The CSS classes of the widget.
+        disabled: Whether the widget is disabled or not.
+
+    """
+
+    DEFAULT_CSS = """
+    Digits {
+        width: 1fr;
+        height: auto;
+        text-align: left;
+        text-style: bold;
+        box-sizing: border-box;
+    }
+    """
+
+    def __init__(
+        self,
+        value: str = "",
+        *,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ) -> None:
+        if not isinstance(value, str):
+            raise TypeError("value must be a str")
```

Comment:
This deserves a `Raises` in the docstring.

I also wonder if it's worth actually allowing `str | float`.

Reply:
I considered that, but there are more numbers than floats or ints. There's Decimal etc. We could just `str` all the things, and allow `Any`, but that didn't sit well.

By accepting `str` it doesn't limit what you can display, and it makes it clear that the dev needs to format it.